development.md

```diff
@@ -0,0 +1,79 @@
+# Development Guideline
+
+This document is the canonical source of truth for tools, dependencies and toolchain used to develop KUDO.
+
+Please submit an issue if you:
+* Notice a requirement that this doc does not capture.
+* Find a different doc that specifies requirements (the doc should instead link here).
+
+
+## Prerequisites
+
+* All interactions including code and comments need to follow the [code of conduct](code-of-conduct.md)
+* All Contributions must follow the [contribution guideline](CONTRIBUTING.md)
+
+The contributions guideline provides details on copyright and process.  This document focuses on technical toolchain and local development.
+
+## Toolchain
+
+The follow are a list of tools needed in order to build, run, test and debug locally.
```

Comment:
Nit: Most of these tools aren't needed but indeed very useful. Should we differ between required (Go, git, unix commands used in our scripts) and useful (goreleaser for team members doing releases, jq, yq)?

Reply:
the tools mentioned are used in some way in our make, build, test or run scripts.   These are not just a grouping of useful tools.

Reply:
well... now that I said that and I'm reviewing again... goreleaser is not in a script... but it is noted as used for our release process.  I'm not sure how to organize based on your feedback...   all of these are required for some function in the dev cycle / make targets / scripts.

Reply:
`minikube`, `jq`, `yq`,  aren't used in our scripts. We also need to mention the `kubebuilder` binaries (`kubebuilder`, `kube-apiserver`, `etcd`) that are expected by the integration tests if one wants to run these tests locally

Reply:
But let's just say that all of these tools are useful, but not all of them are needed in the dev cycle.

Reply:
`jq` and `yq` are used in `hack/update-manifests.sh`..  but good points

Reply:
Oh! You're right, my local copy wasn't up-to-date with latest `main`, my bad :)